### PR TITLE
sg: fix ops update-images auth bug

### PR DESCRIPTION
**Pull request by jhchabran**
This small PR fixes a bug with the authentication flags, who weren't passed down to the functions doing the work. 

I added some clarification around an error message along the way, which helps to nail down which image it causing a problem. 

## Test plan

- Ran it locally and tested manually entered credentials.

### CODE REVIEW

File: dev/sg/internal/images/images.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ func ParseHelm(path string, creds credentials.Credentials, pinTag string) error
 	}
 
 	var values map[string]interface{}
-	json.Unmarshal(rawValues, &values)
+	_ = json.Unmarshal(rawValues, &values)
```
Comment (danieldides) [inline dev/sg/internal/images/images.go:109]: I know the odds of this failing are slim-to-none but should we still catch and log this error?

Comment (jhchabran) [reply dev/sg/internal/images/images.go:109]: I hesitated a bit on this one, but as we're getting json from the call just a bit above, which is going to call marshall as well, I ruled it out as okay to skip while also silencing the linter warning. I honestly have no strong opinion on this. Let's just catch it we don't ever ask ourselves this question again :P

File: dev/sg/sg_ops.go [hunk 2]
```diff
@@ -68,19 +68,22 @@ func opsUpdateImage(ctx context.Context, args []string) error {
...
 			// We do not want any error handling here, just fallback to anonymous requests
 			writeWarningLinef("Registry credentials are not provided and could not be retrieved from docker config.")
 			writeWarningLinef("You will be using anonymous requests and may be subject to rate limiting by Docker Hub.")
-			dockerCredentials = &credentials.Credentials{ServerURL: "https://index.docker.io/v1/", Username: "", Secret: ""}
+			dockerCredentials.Username = ""
+			dockerCredentials.Secret = ""
 		} else {
 			writeFingerPointingLinef("Using credentials from docker credentials store (learn more https://docs.docker.com/engine/reference/commandline/login/#credentials-store)")
-			opsUpdateImagesContainerRegistryUsernameFlag = &dockerCredentials.Username
-			opsUpdateImagesContainerRegistryPasswordFlag = &dockerCredentials.Secret
```
Comment (bobheadxi) [inline dev/sg/sg_ops.go:83]: Should this be removed? It looks like a backcompat measure for code that might use the flags directly (though I guess we do want to discourage that)

**Review by bobheadxi — APPROVED** [review]
LGTM, but might be good to have Michael take a look too since he wrote the credentials plugin stuff here!

**Comment by jhchabran** [comment]
Hey @michaellzc, any chance you can take a brief look at this?

**Review by michaellzc — APPROVED** [review]
lgtm

sorry for the late review. I'm horrible at keeping up github notification

**Pull request merged**